### PR TITLE
Make flashcard viewer answer buttons respect safe display setting

### LINKED ISSUE #7128: Disable button animation
Feature request

In Ankidroid 2.13 the buttons animate which slows down the flow. I prefer the practice session to be snappy and the animations are a distraction. I'd like this feature to be switched off. 

###### Reproduction Steps

1. 
2. 
3. 

###### Debug info
Refer to the [support page](https://ankidroid.org/docs/help.html) if you are unsure where to get the "debug info".

###### Research
*Enter an [x] character to confirm the points below:*

- [ ] I have read the [support page](https://ankidroid.org/docs/help.html) and am reporting a bug or enhancement request specific to AnkiDroid
- [ ] I have checked the [manual](https://ankidroid.org/docs/manual.html) and the [FAQ](https://github.com/ankidroid/Anki-Android/wiki/FAQ) and could not find a solution to my issue
- [ ] I have searched for similar existing issues here and on the user forum
- [ ] (Optional) I have confirmed the issue is not resolved in the latest alpha release ([instructions](https://docs.ankidroid.org/manual.html#betaTesting))

### PR DESCRIPTION
**Pull request by unusualstranger**
## Purpose / Description
This pull request makes disabling ripples on answer buttons possible by enabling Safe Display option.

## Fixes
Fixes #7128 

## Approach
This pull request separates ripple (v21) and non-ripple drawables into different resource IDs so referencing them from runtime becomes possible. Then, on activity creation, SDK version and safe display option is checked and if checks pass then code sets ripple backgrounds.

## How Has This Been Tested?

I've ran the app on Android 9 physical device (HUAWEI Honor 9, STF-AL10, EMUI 9) and on AVD API 16. I haven't experienced any crashes or bugs during my tests.

I'm ill right now (→ my current health state is not perfect) so there is a big probability that I've forgot about something. You can review my changes and comment/push commits if necessary.

## Checklist
_Please, go through these checks before submitting the PR._

- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [ ] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/anki/Previewer.java [hunk 1]
```diff
@@ -258,6 +267,8 @@ public void onClick(View v) {
     };
 
     private void updateButtonsState() {
+        mPreviewToggleAnswerText.setText(mShowingAnswer ? R.string.hide_answer : R.string.show_answer);
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/anki/Previewer.java:270]: did this change need to occur?

File: AnkiDroid/src/main/java/com/ichi2/anki/Reviewer.java [hunk 2]
```diff
@@ -703,11 +703,21 @@ protected void displayAnswerBottomBar() {
         // Set correct label and background resource for each button
         // Note that it's necessary to set the resource dynamically as the ease2 / ease3 buttons
         // (which libanki expects ease to be 2 and 3) can either be hard, good, or easy - depending on num buttons shown
-        final int[] background = Themes.getResFromAttr(this, new int [] {
-                R.attr.againButtonRef,
-                R.attr.hardButtonRef,
-                R.attr.goodButtonRef,
-                R.attr.easyButtonRef});
+        int[] backgroundIds;
+        if (Build.VERSION.SDK_INT >= 21 && animationEnabled()) {
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/anki/Reviewer.java:707]: might want to extract this conditional to a method in the `AbstractFlashcardViewer`, there's a bit of repetition

21 should be `Build.VERSION_CODES.LOLLIPOP`

**Review by david-allison-1 — CHANGES_REQUESTED** [review]
This would have taken me about 20x longer to figure out. Thank you so much ❤❤❤

Minor nitpicks, nothing major, just pulling this into my emulators for a final test

**Comment by david-allison-1** [comment]
All looks good from a testing point of view.

File: AnkiDroid/src/main/java/com/ichi2/anki/Previewer.java (hunk 1 above)
Comment (unusualstranger) [reply AnkiDroid/src/main/java/com/ichi2/anki/Previewer.java:270]: There was a bug: when card count is 1, this function returns too early and doesn't update flip button's label, so when an user would flip the card, button's label would stay the same. I've fixed that but that's a bad practice to smash two fixes into one PR, sorry. Should I take it out or it's OK?

**Review by david-allison-1 — APPROVED** [review]
Let's get this in. Nitpicks can wait

File: AnkiDroid/src/main/java/com/ichi2/anki/Previewer.java (hunk 1 above)
Comment (david-allison-1) [reply AnkiDroid/src/main/java/com/ichi2/anki/Previewer.java:270]: Nope - that's fine

**Review by mikehardy — APPROVED** [review]
This looks good to me too - sorry to hear you are not feeling well! Thanks for working on this despite feeling not great. I'll be able to include in 2.13.1 which will probably go out today

**Pull request merged**